src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2641,9 +2641,9 @@ private bool IsOperandErrors(CSharpSyntaxNode node, ref BoundExpression operand,
                     if (!operand.HasAnyErrors)
                     {
                         Error(diagnostics, ErrorCode.ERR_LambdaInIsAs, node);
-                        operand = BadExpression(node, operand).MakeCompilerGenerated();
                     }
 
+                    operand = BadExpression(node, operand).MakeCompilerGenerated();
```

Comment:
Why are we doing this unconditionally now?

Reply:
Because it is unconditionally illegal to have a lambda or method group here. This line just gives the expression an error type so later code doesn't see a typeless expression.

---
In reply to: [180249518](https://github.com/dotnet/roslyn/pull/26035#discussion_r180249518) [](ancestors = 180249518)